app/code/Magento/InventorySales/Block/Adminhtml/Order/View/Tab/SourceSelection.php

```diff
@@ -0,0 +1,163 @@
+     * @var SourceRepositoryInterface
+     */
+    private $sourceRepository;
+    /**
+     * @var ShippingAlgorithmInterface
+     */
+    private $shippingAlgorithm;
+
+    /**
+     * @param Context $context
+     * @param Registry $registry
+     * @param SearchCriteriaBuilder $searchCriteriaBuilder
+     * @param SourceItemRepositoryInterface $sourceItemRepository
+     * @param SourceRepositoryInterface $sourceRepository
+     * @param ShippingAlgorithmInterface $shippingAlgorithm
+     * @param array $data
+     */
+    public function __construct(
+        Context $context,
+        Registry $registry,
+        SearchCriteriaBuilder $searchCriteriaBuilder,
+        SourceItemRepositoryInterface $sourceItemRepository,
+        SourceRepositoryInterface $sourceRepository,
+        ShippingAlgorithmInterface $shippingAlgorithm,
+        array $data = []
+    ) {
+        parent::__construct($context, $data);
+        $this->registry = $registry;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+        $this->sourceItemRepository = $sourceItemRepository;
+        $this->sourceRepository = $sourceRepository;
+        $this->shippingAlgorithm = $shippingAlgorithm;
+    }
+
+    /**
+     * Getting source items for order
+     *
+     * @return array
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    public function getSourceItemsData(): array
+    {
+        $order = $this->registry->registry('current_order');
+
+        $sourceSelectionMap = $this->shippingAlgorithm->get($order)
+                                                    ->getSourceSelections();
+
+        $sourceItemsData = [];
+        foreach ($order->getAllVisibleItems() as $orderItem) {
+            $itemSku = $orderItem->getSku();
+            $sourceItems = $this->getSourceItemsBySku($orderItem->getSku());
+
+            $itemSourceSelections = [];
+            /** @var SourceSelectionInterface[] $sourceSelections */
+            $sourceSelections = $sourceSelectionMap[$itemSku];
+            foreach ($sourceSelections as $sourceSelection) {
+                $itemSourceSelections[$sourceSelection->getSourceCode()] = $sourceSelection->getQty();
+            }
+
+            foreach ($sourceItems as $sourceItem) {
+                $sourceCode = $sourceItem->getSourceCode();
+                $sourceName = $this->sourceRepository->get($sourceCode)->getName();
+                $deductedQty = $itemSourceSelections[$sourceCode] ?? 0;
+                $sourceItemsData[$sourceName][] = [
+                    'sku' => $itemSku,
+                    'qty' => $sourceItem->getQuantity(),
+                    'qty_deducted' => $deductedQty
+                ];
+            }
+        }
+        
+        return $sourceItemsData;
+    }
+
+    /**
+     * @param string $sku
+     * @return SourceItemInterface[]
+     */
+    private function getSourceItemsBySku(string $sku): array
+    {
+        $searchCriteria = $this->searchCriteriaBuilder
```

Comment:
Need to find same code in MSI project
Maybe make sense to create sugar service for avoiding boilerplate code